community/function/src/main/java/org/neo4j/function/IntSupplier.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.function;
+
+/**
+ * Represents a supplier of int-valued results. This is the int-producing primitive specialization of {@link Supplier}.
+ */
+public interface IntSupplier extends ThrowingIntSupplier<RuntimeException>
+{
+    /**
+     * Gets a result.
+     *
+     * @return a result
+     */
+    int getAsInt();
```

Comment:
It's `getAsInt()` because that's what Java does, and we want to switch to Java 8 instead of this sometime. https://docs.oracle.com/javase/8/docs/api/java/util/function/IntSupplier.html#getAsInt--